dev/bots/custom_rules/analyze.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' show Directory;
+
+import 'package:analyzer/dart/analysis/analysis_context.dart';
+import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/analysis/session.dart';
+import 'package:path/path.dart' as path;
+
+import '../utils.dart';
+
+/// Analyzes the given `workingDirectory` with the given set of [AnalyzeRule]s.
```

Comment:
This doesn't work for any workingDirectory, though. The working directory must contain the flutter package at `package/flutter/lib`. Typically we call this directory the flutter root, so I would just call this parameter flutterRoot (or actually make analyzeDirectoryWithRules work on any directory).